src/hooks/useBreakpoint.tsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+import { BreakpointKey, Breakpoints } from "../Css";
+
+/**
+ * A React hook to return a record of responsive breakpoints that updates on resize.
+ *
+ * @example
+ * const { breakpoints } = useBreakpoint();
+ * if(breakpoints.mdAndDown) {...do something cool}
+ */
+
+type BreakpointsType = Record<BreakpointKey, boolean>;
+
+export function useBreakpoint(): { breakpoints: BreakpointsType } {
+  const [breakpoints, setBreakpoints] = useState({} as BreakpointsType);
+
+  useEffect(() => {
+    function handleResize() {
+      // Set breakpoints on resize
+      let bps = {} as BreakpointsType;
+      Object.entries(Breakpoints).forEach(([name, bp]) => {
+        bps[name as keyof BreakpointsType] = window.matchMedia(bp.replace("@media screen and ", "")).matches;
```

Comment:
Thought: Kinda looks like leaving in `screen and` will still work:

```
window.matchMedia("screen and (min-width:1px)")
MediaQueryList {media: 'screen and (min-width: 1px)', matches: true, onchange: null}
```

So maybe `@media` should be the only thing we strip?

Reply:
good call - updated